### PR TITLE
Disco test

### PR DESCRIPTION
**Pull request by ngrayson**
added end to end testing via a second discord bot, and the Prompt and ActionResponse classes
added a few end to end tests

### CODE REVIEW

File: chatbot/__test__/emojiManager.spec.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe('Giha Discord Interface - emojiManager', () => {
```
Comment (ngrayson) [inline chatbot/__test__/emojiManager.spec.js:1]: maybe don't need

File: chatbot/botManager.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        chatbotReady = true
+        log('bot.commands:')
+        bot.commands.forEach((command) => {
+          log(command.help.name)
+        })
+        timeManager()
+        resolve(bot)
+      })
+
+      bot.on('message', async (msg) => {
```
Comment (ngrayson) [inline chatbot/botManager.js:47]: check if async is needed

File: chatbot/discoTest/endToEndTests.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  let testSuite = {
+    suiteName: 'Mock Test Suite',
+    tests: tests,
+  }
+  testSuites.push(testSuite)
+  return testSuites
+}
+
+// begin Tests
+try {
```
Comment (ngrayson) [inline chatbot/discoTest/endToEndTests.js:149]: should be an exported fuction

File: chatbot/botManager.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import { log } from '../util/util.js'
```
Comment (ngrayson) [inline chatbot/botManager.js:1]: file should be called chatBot

**Review by rabelfish — APPROVED** [review]
LGTM

**Pull request merged**